provisioning.go

```diff
@@ -388,6 +406,47 @@ func (prov *ProvisioningAPI) checkSessionAndReturnHandle(ctx context.Context, w
 	return handle
 }
 
+func (prov *ProvisioningAPI) WhoAmI(w http.ResponseWriter, r *http.Request) {
+	user := r.Context().Value(provisioningUserKey).(*User)
+	log := prov.log.With().
+		Str("action", "whoami").
+		Str("user_id", user.MXID.String()).
+		Logger()
+	log.Debug().Msg("getting whoami")
+
+	status := http.StatusOK
+	data := &WhoAmIResponse{
+		Permissions: int(user.PermissionLevel),
+		MXID:        user.MXID.String(),
+	}
+	if user.IsLoggedIn() {
+		puppet := user.bridge.GetPuppetBySignalID(user.SignalID)
+		if puppet == nil {
+			status = http.StatusInternalServerError
+			data.Signal = &WhoAmIResponseSignal{
+				Number: user.SignalUsername,
+				Ok:     false,
+				Error:  "Failed to get puppet by Signal ID",
+			}
+			log.Error().
+				Str("signal_id", user.SignalID.String()).
+				Msg(data.Signal.Error)
+		} else {
+			data.Signal = &WhoAmIResponseSignal{
+				Number: user.SignalUsername,
+				UUID:   user.SignalID.String(),
+				Name:   puppet.Name,
+				Ok:     true,
+			}
+		}
+	}
+	jsonResponse(w, status, Response{
+		Success:        true,
+		Status:         "ok",
+		WhoAmIResponse: data,
```

Comment:
The `Response` struct shouldn't be used for anything other than login (resolve identifier was just implemented wrong)